crates/core/executor/src/executor.rs

```diff
@@ -190,11 +192,52 @@ impl<'a> Executor<'a> {
         Self::with_context(program, opts, SP1Context::default())
     }
 
+    /// Crete a new runtime for the program, and setup the profiler if `TRACE_FILE` env var is set
+    /// and the feature flag `profiling` is enabled.
+    #[must_use]
+    pub fn with_context_and_elf(
+        opts: SP1CoreOpts,
+        context: SP1Context<'a>,
+        elf_bytes: &[u8],
+    ) -> Self {
+        let program = Program::from(elf_bytes).expect("Failed to create program from ELF bytes");
+
+        #[cfg(not(feature = "profiling"))]
+        return Self::with_context(program, opts, context);
+        
+        #[cfg(feature = "profiling")]
+        {
+            let mut this = Self::with_context(program, opts, context);
+        
+            let trace_buf = std::env::var("TRACE_FILE").ok().map(|file| {
+                let file = File::create(file).unwrap();
+                BufWriter::new(file)
+            });
+            
+            if let Some(trace_buf) = trace_buf {
+                println!("Profiling enabled");
+
+                let sample_rate = std::env::var("TRACE_SAMPLE_RATE")
+                    .ok()
+                    .and_then(|rate| {
+                        println!("Profiling sample rate: {rate}");
+                        rate.parse::<u32>().ok()
+                    })
+                    .unwrap_or(1);
+
+                this.profiler = Some((
+                    Profiler::new(elf_bytes, sample_rate as u64).expect("Failed to create profiler"),
+                    trace_buf,
+                ));
+            }
+
+            this
+        }
+    }
+
     /// Create a new runtime from a program, options, and a context.
     ///
-    /// # Panics
-    ///
-    /// This function may panic if it fails to create the trace file if `TRACE_FILE` is set.
+    /// Note: This function *will not* set up the profiler.
     #[must_use]
```

Comment:
to clarify, the purpose of this function now is solely to maintain backwards compatibility?